src/pyscheduling/SMSP/SingleMachine.py

```diff
@@ -440,485 +448,119 @@ def toJSON(self):
-                    setupTime = instance.S[job_prev_i][job_i]
-                else:
-                    setupTime = 0
-                proc_time = instance.P[job_i]
-                ci = startTime + setupTime + proc_time
-                wiCi += instance.W[job_i]*ci
+            startIndex (int, optional): The index from which we start fixing the schedule. Defaults to 0.
+            obj_cache (list[int]): The objective's cache, it can be wiCi, wiTi or other. Defaults to None.
 
-            job_prev_i = job_i
-
-        return wiCi
-
-    def total_weighted_completion_time_swap(self, pos_i: int, pos_j: int, instance: SingleInstance):
-        """Computes the machine's total weighted completion time if we insert swap jobs at position "pos_i" and "pos_j"
-        in the machine's job_schedule
-        Args:
-            pos_i (int): position of the first job to be swapped
-            pos_j (int): position of the second job to be swapped
-            instance (SingleInstance): the current problem instance
         Returns:
-            int : total weighted completion time
+            tuple: (startIndex, obj_cache) 
         """
-        first_pos = min(pos_i, pos_j)
-
-        ci = 0
-        if pos_i == 0:
-            job_prev_i = self.job_schedule[pos_j]
-        else:
-            job_prev_i = self.job_schedule[pos_i]
-        wiCi = 0
-        if first_pos > 0:  # There's at least one job in the schedule
-            ci = self.job_schedule[first_pos - 1].end_time
-            job_prev_i = self.job_schedule[first_pos - 1].id
-            wiCi = self.wiCi_index[first_pos - 1]
-
-        for i in range(first_pos, len(self.job_schedule)):
-
-            if i == pos_i:  # We take pos_j
-                job_i = self.job_schedule[pos_j][0]  # (Id, startTime, endTime)
-            elif i == pos_j:  # We take pos_i
-                job_i = self.job_schedule[pos_i][0]
-            else:
-                job_i = self.job_schedule[i][0]  # Id of job in position i
-
-            if hasattr(instance, 'R'):
-                startTime = max(ci, instance.R[job_i])
-            else:
-                startTime = ci
-            if hasattr(instance, 'S'):
-                setupTime = instance.S[job_prev_i][job_i]
-            else:
-                setupTime = 0
-            proc_time = instance.P[job_i]
-            ci = startTime + setupTime + proc_time
-            wiCi += instance.W[job_i]*ci
-
-            job_prev_i = job_i
-
-        return wiCi
-
-    def total_weighted_lateness(self, instance: SingleInstance, startIndex: int = 0):
-        """Fills the job_schedule with the correct sequence of start_time and completion_time of each job and returns the total weighted lateness
+        objective = instance.get_objective()
+        obj_cache = self.objectives_map.get(objective, -1)
+        if obj_cache == -1: # No cache is used
+            return startIndex, None
+        if obj_cache is None:  # Iniates wiCi_index to the size of job_schedule
```

Comment:
I think, for the comment message in green, we have to replace wiCi_index by obj_cache !

Reply:
I think, for the comment message in green, we have to replace wiCi_index by obj_cache !

Reply:
Yes I'll change it